index.js

```diff
@@ -319,28 +322,42 @@ module.exports = {
     }
 
     function listMembers(groupId, opts = {}) {
+      const { live } = opts
+
+      if (!live) {
+        const deferredSource = pullDefer.source()
+
+        getPreferredEpoch(groupId, (err, epoch) => {
+          if (err)
+            return deferredSource.abort(
+              clarify('failed to load preferred epoch')
+            )
+
+          const source = pull(
+            getMembers.stream(epoch.id),
+            // if it's not live, only include added entries
+            pull.filter((update) => update.added)
+          )
+          deferredSource.resolve(source)
+        })
+
+        return deferredSource
+      }
+
       return pull(
-        getPickedEpoch(groupId, { live: !!opts?.live }),
-        pull.map((pickedEpoch) => {
-          console.log('pickedEPoch', pickedEpoch)
-          return getMemberUpdates(pickedEpoch.id, { live: !!opts?.live })
-        }),
-        pull.flatten(),
-        pull.map((update) => {
-          if (update.added) return update.added
-          return update
-        }),
-        // if it's not live, only include adds
-        pull.filter((update) => opts?.live || !update.excluded)
+        getPreferredEpoch.stream(groupId, { live }),
+        pull.map((epoch) => getMembers.stream(epoch.id, { live })),
+        pullFlatMerge()
+        // NOTE pullFlatMerge needed in live case otherwise new streams don't get a word in!
```

Comment:
this was my solution to the "oh no the last strep didn't close" thing
Its sucky but I think we need to leave the streams open to keep hearing the exclusions.

I kinda hate we have this live stream... it has been so much work to get sorted and I'm still not confident with how it will behave in edge cases : 
- what happens when there are multiple exclusions happen for a epoch (you an I fork the group) ... this live stream should do what?
- in the case of a fork ... what members am I getting?

Safest thing to do would be to make this not live IMO.
Until we need live streaming members. In my experience very few things need live streaming, and even then I tend to use the live stream more as a trigger to run some async process. MUCH simpler.

What do you think?

Reply:
We need livestreaming for replication scheduler https://github.com/ssbc/ssb-replication-scheduler/blob/2c5133150ae680c60981e4e2f783365ee83a95a2/index.js#L82

Reply:
> what happens when there are multiple exclusions happen for a epoch (you an I fork the group) ... this live stream should do what?

instead of emitting smaller chunks of state updates (like we do now) maybe it's simpler to go with my previous suggestion of `Stream<ListOfAllCurrentMembers[]>` where it's very clear what the reduced state is?

Reply:
Good idea - I like this much better